tableaudocumentapi/datasource.py

```diff
@@ -235,6 +235,56 @@ def clear_repository_location(self):
             self._datasourceXML.remove(tag)
 
     ###########
+    # Folders
+    ###########
+
+    @property
+    def folders(self):
+        """ Returns all folders.
+        """
+        return self._datasourceTree.findall('.//folder')
+
+    def add_folder(self, name, role, fields):
+        """ Adds a folder field with the given fields.
+
+        Args:
+            name:  Name of the new folder. String.
+            role:  Role of the new folder. String.
+            fields:  Fields of the new folder. String.
+
+        Returns:
+            The new calculated folder that was created. ET.Element.
+        """
+        # TODO: It might be better to create a dedicated "Folder" object:
```

Comment:
Thanks Tyler, I'm planning on doing exactly that. It's actually one of the bigger parts (time/effort-wise) of the implementation. 
Related question: Does the API also support Tableau 9 or earlier Workbooks/Datasources? Currently not sure if I need to take this into consideration and test for it.